components/common/src/command/package/install.rs

```diff
@@ -702,11 +704,38 @@ impl<'a> InstallTask<'a> {
     where
         T: UIWriter,
     {
-        artifact.unpack(Some(self.fs_root_path))?;
-        ui.status(Status::Installed, artifact.ident()?)?;
+        let ident = &artifact.ident()?;
+        let real_install_path = &pkg_install_path(ident, Some(self.fs_root_path));
+        let real_install_base = Path::new(real_install_path)
+            .parent()
+            .ok_or_else(|| Error::PackageUnpackFailed("Could not determine parent directory for package unpack location".to_owned()))?;
+        let temp_dir = self.temporary_unpack_directory(real_install_base, real_install_path)?;
+        let temp_install_path = &pkg_install_path(ident, Some(temp_dir.path()));
+        artifact.unpack(Some(temp_dir.path()))?;
+        fs::rename(temp_install_path, real_install_path)?;
+        self.fsync_dir(real_install_base)?;
+        ui.status(Status::Installed, ident)?;
         Ok(())
     }
 
+
+    fn fsync_dir(&self, dir: &Path) -> Result<()> {
+        // TODO(ssd) 2018-07-23: This works on Linux but I'm not sure
+        // how portable it is.
```

Comment:
I think you can remove this comment. There is [a Windows implementation](https://github.com/rust-lang/rust/blob/master/src/libstd/sys/windows/fs.rs#L277). In general, the rust documentation calls out if something is platform-specific.